src/repo/ShipItRepoGIT.php

```diff
@@ -90,12 +90,15 @@ private static function parseHeader(string $header): ShipItChangeset {
     list($envelope, $message) = explode("\n\n", trim($header), 2);
 
     $message = trim($message);
-    if (strpos($message, "\n---\n") !== false) {
-      // Get rid of the file list
-      $parts = explode("\n---\n", $message);
-      array_pop($parts);
-      $message = implode("\n---\n", $parts);
+
+    $matches = [];
+    preg_match("/^((.*\\n)?)---\\n /us", $message, $matches, PREG_OFFSET_CAPTURE);
```

Comment:
lint: line too long (should be 80 chars)

Sorry, we're in a bad state for running our linters on hack/php code externally :(